File: easy/3069-distribute-elements-into-two-arrays-i/solution.py
from typing import List


class Solution:
    def resultArray(self, nums: List[int]) -> List[int]:
        """
        3069. Distribute Elements Into Two Arrays I
        Time: O(n)
        Space: O(n)
        """
        arr1 = [nums[0]]
        arr2 = [nums[1]]

        for x in nums[2:]:
            if arr1[-1] > arr2[-1]:
                arr1.append(x)
            else:
                arr2.append(x)

        return arr1 + arr2

File: easy/3069-distribute-elements-into-two-arrays-i/solution.java
import java.util.ArrayList;
import java.util.List;

/**
 * 3069. Distribute Elements Into Two Arrays I
 * Time: O(n)
 * Space: O(n)
 */
class Solution {
    public int[] resultArray(int[] nums) {
        List<Integer> arr1 = new ArrayList<>();
        List<Integer> arr2 = new ArrayList<>();
        arr1.add(nums[0]);
        arr2.add(nums[1]);

        for (int i = 2; i < nums.length; i++) {
            if (arr1.get(arr1.size() - 1) > arr2.get(arr2.size() - 1)) {
                arr1.add(nums[i]);
            } else {
                arr2.add(nums[i]);
            }
        }

        int[] result = new int[nums.length];
        int idx = 0;
        for (int v : arr1) {
            result[idx++] = v;
        }
        for (int v : arr2) {
            result[idx++] = v;
        }
        return result;
    }
}
